[TA-5032]: ibc testsuite

GuillemGarciaDev opened the pull request:
# [TA-5032]: ibc testsuite

## Changes :hammer_and_wrench:

### modules/cosmos

- `ibc` transfer testsuite between IBC chains
- test sending native assets from source chain to destination chain.

## Notes

`EIP712` signatures are not supported in `@cosmjs`. Need to implement it custom.

GuillemGarciaDev committed:
feat(cosmos): adaptint `@cosmjs` to support `EthAccount`

GuillemGarciaDev committed:
feat(cosmos): template mainnet config

banasa44 commented on modules/cosmos/src/modules/ibc/secp256k1.ts:
should we move this file into `packages/shared`?

banasa44 reviewed (commented):
`modules/ibc` could be refactored, structure is currently highly packed. Suggestion:
- `/core` with interfaces, errors and constants
- `/parsers`
- `/utils` multiple utility functions from modules/ibc
- `/services` (`/providers` ? not sure about this one, maybe add more paths) for business logic orchestration, clients and signers. Or `/clients` and leave `signer.ts` at top level.

banasa44 commented on modules/cosmos/src/modules/ibc/signer.ts:
```diff
@@ -0,0 +1,364 @@
+import { SignDoc } from "cosmjs-types/cosmos/tx/v1beta1/tx";
+import { encodeSecp256k1Signature } from "./secp256k1";
+
+interface AccountDataWithPrivkey extends AccountData {
```
don't forget to complete

AdriaCarrera committed:
Merge branch 'main' into cosmos/feat/ibc-testsuite

# Conflicts:
#	modules/cosmos/configs/mainnet.module.config.json
#	modules/cosmos/configs/testnet.module.config.json
#	modules/cosmos/module.config.json
#	modules/cosmos/package.json
#	pnpm-lock.yaml

AdriaCarrera committed:
feat: update Cosmos network configurations and dependencies

- Updated RPC URLs to use secure HTTPS endpoints across all environments.
- Added IBC module and configuration for chain interoperability.
- Revised EVM configurations with proper `evmDenom`, active precompiles, and enhanced fee market parameters.
- Introduced Cosmos SDK v0.33.1 dependencies for compatibility improvements.
- Included devnet configurations and adjusted test cases for mainnet, testnet, and devnet execution.
- Made adjustments in `pnpm-lock.yaml` for upgraded dependencies and package resolution.
